botocore/parsers.py

```diff
@@ -722,6 +726,39 @@ def _initial_body_parse(self, xml_string):
         return self._parse_xml_string_to_dom(xml_string)
 
 
+class JSONParser(BaseJSONParser):
+
+    EVENT_STREAM_PARSER_CLS = EventStreamJSONParser
+
+    """Response parser for the "json" protocol."""
+    def _do_parse(self, response, shape):
+        parsed = {}
+        if shape is not None:
+            event_name, event_shape = self._get_event_stream_member(shape)
```

Comment:
Yep, this is unfortunate. For now I've moved the logic to find this into the models which gives us caching at the client level. To avoid additional logic in the parsers I've added it to both the base `Shape` and the `Structure`, so short of a modelling change I think the impact of this should be minimal now.